### PR TITLE
Expose configuration of HTTP Connector proxy support

### PR DESCRIPTION
**Pull request by v1-evansb**
We are expecting that this Docker image runs behind a reverse-proxy.

Some other infrastructure component handles TLS termination. This could be something like haproxy, apache httpd, nginx, or an ELB.

We want to configure the Connector so that Tomcat is aware of this configuration. Thus [Tomcat will know that TLS is being used, and on which port](https://tomcat.apache.org/tomcat-8.5-doc/config/http.html#Proxy_Support).

See https://willwarren.com/2014/01/27/running-apache-tomcat-with-ssl-behind-amazon-elb/

We are adding two new environment variables to the Dockerfile:

* TOMCAT_CONNECTOR_PROXY_PORT = 443
* ENV TOMCAT_CONNECTOR_SCHEME = https

This is a secure profile by default, suitable for running in production behind a proxy that terminates TLS.

We are including these in the `CATALINA_OPTS` environment variable that the image defines.

```sh
ENV CATALINA_OPTS="$CATALINA_OPTS \
  -Dconnector.proxy.port=$TOMCAT_CONNECTOR_PROXY_PORT \
  -Dconnector.scheme=$TOMCAT_CONNECTOR_SCHEME"
```

The properties `connector.proxy.port` & `connector.scheme` are referenced in
the server.xml file:

```xml
<Connector
  port="8080"
  protocol="HTTP/1.1"
  proxyPort="${connector.proxy.port}"
  scheme="${connector.scheme}"/>
```

Alternatively you can specify overrides of the environment variables with runtime arguments:

```sh
docker container run -e TOMCAT_CONNECTOR_PROXY_PORT=8080 -e TOMCAT_CONNECTOR_SCHEME=http myhardenedcontainer
```

### CODE REVIEW

**Comment by v1-evansb** [comment]
🆘 I added usage instructions for images based of this one, so that the CATALINA_OPTS are set properly
https://github.com/ministryofjustice/docker-tomee/blob/feature/LM-1353-connector-proxy-support/README.md#usage
I was not entirely happy with that, but I could not find another way to inject the new  env vars in the base image:
```
CMD CATALINA_OPTS="$CATALINA_OPTS \
    -Dconnector.proxy.port=$TOMCAT_CONNECTOR_PROXY_PORT \
    -Dconnector.scheme=$TOMCAT_CONNECTOR_SCHEME" \
    catalina.sh run
```
I know that we will be overriding the CMD in each application's Dockerfile, so I thought it was a reasonable approach... if there is a better way to do this please let me know.

File: server.xml [hunk 1]
```diff
@@ -68,9 +68,22 @@
...
+      <Connector
+              port="8080"
+              protocol="HTTP/1.1"
+              connectionTimeout="20000"
+              redirectPort="8443"
+              xpoweredBy="false"
+              server=""
+              proxyPort="${connector.proxy.port}"
+              scheme="${connector.scheme}"
+              secure="true"
```
Comment (jabley) [inline server.xml:84]: Should this also allow injection via a property?

**Review by jabley — DISMISSED** [review]
Will tidy up a couple of things in a later change. Approving now to unblock Barry.

**Review by jabley — APPROVED** [review]

File: server.xml (hunk 1 above)
Comment (jabley) [reply server.xml:84]: This was addressed in #7.

**Pull request merged**